MergerLogicUnitTests/ImageProcessing/TileScalerTest.cs

```diff
@@ -0,0 +1,66 @@
+﻿using MergerLogic.Batching;
+using MergerLogic.DataTypes;
+using MergerLogic.ImageProcessing;
+using MergerLogic.Monitoring.Metrics;
+using Microsoft.Extensions.Logging;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+using System;
+using System.Collections.Generic;
+using System.IO;
+
+namespace MergerLogicUnitTests.ImageProcessing
+{
+    [TestClass]
+    [TestCategory("unit")]
+    [TestCategory("imageProcessing")]
+    [DeploymentItem(@"../../../ImageProcessing/TestImages")]
+    public class TileScalerTest
+    {
+
+        #region mocks
+
+        private MockRepository _mockRepository;
+
+        private TileScaler _testTileScaler;
+
+        #endregion
+
+        [TestInitialize]
+        public void BeforeEach()
+        {
+            this._mockRepository = new MockRepository(MockBehavior.Loose);
+
+            var metricsProviderMock = this._mockRepository.Create<IMetricsProvider>();
+            var tileScalerLoggerMock = this._mockRepository.Create<ILogger<TileScaler>>();
+
+            this._testTileScaler = new TileScaler(metricsProviderMock.Object, tileScalerLoggerMock.Object);
+        }
+
+        #region Upscale
+
+
+        public static IEnumerable<object[]> GetUpscaleTilesTestParameters()
+        {
+            yield return new object[] {
+                File.ReadAllBytes("3.jpeg"),
+                new Coord(9, 0, 0),
+                new Coord(10, 0, 0),
+                File.ReadAllBytes("3_upscaled_9_10.jpeg"),
+            };
+        }
+        [TestMethod]
+        [TestCategory("Upscale")]
+        [DynamicData(nameof(GetUpscaleTilesTestParameters), DynamicDataSourceType.Method)]
+        public void Upscale(byte[] tileBytes, Coord baseTileCoord, Coord targetCoord, byte[] expectedTileBytes)
```

Comment:
Could you please add a test for upscaling a tile more than 8 zoom levels? As I mentioned before this is a special case we should check.

Reply:
added a test case

Reply:
Great!